Darwin/Services/StitchRemoteMongoDBService/StitchRemoteMongoDBServiceTests/SyncIntTests.swift

```diff
@@ -15,7 +15,7 @@ import StitchCoreLocalMongoDBService
 
 private let waitTimeout = UInt64(1e+10)
 
-private extension Document {
+internal extension Document {
```

Comment:
[minor] Since these extensions and classes are no longer unique to this file, we should throw them them into separate files. It'll be easier to keep track of them.

The convention for naming extension files is to do stuff like `Document+Sorted.swift`, `RemoteMongoCollection+SynchronousCRUD.swift`.

StreamJoiner and SyncTestContext can be named normally.